components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -0,0 +1,1955 @@
+     */
+    public MetaFederatedAuthenticator getMetaFederatedAuthenticator(String id) {
+
+        String authenticatorName = base64URLDecode(id);
+        MetaFederatedAuthenticator authenticator = null;
+        try {
+            FederatedAuthenticatorConfig[] authenticatorConfigs =
+                    IdentityProviderServiceHolder.getIdentityProviderManager()
+                            .getAllFederatedAuthenticators();
+            if (authenticatorConfigs != null) {
+                for (FederatedAuthenticatorConfig authenticatorConfig : authenticatorConfigs) {
+                    if (StringUtils.equals(authenticatorConfig.getName(), authenticatorName)) {
+                        authenticator = createMetaFederatedAuthenticator(authenticatorConfig);
+                        break;
+                    }
+                }
+            }
+            return authenticator;
+        } catch (IdentityProviderManagementException e) {
+            throw handleIdPException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_META_AUTHENTICATOR, id);
+        }
+    }
+
+    /**
+     * Get meta information about Identity Provider's outbound provisioning connectors.
+     *
+     * @return List of meta outbound provisioning connectors.
+     */
+    public List<MetaOutboundConnectorListItem> getMetaOutboundConnectors() {
+
+        List<MetaOutboundConnectorListItem> metaOutboundConnectors = null;
+        try {
+            ProvisioningConnectorConfig[] connectorConfigs = IdentityProviderServiceHolder.getIdentityProviderManager()
+                    .getAllProvisioningConnectors();
+            if (ArrayUtils.isNotEmpty(connectorConfigs)) {
+                metaOutboundConnectors = new ArrayList<>();
+                for (ProvisioningConnectorConfig connectorConfig : connectorConfigs) {
+                    MetaOutboundConnectorListItem metaOutboundConnector = createMetaOutboundConnectorListItem
+                            (connectorConfig);
+                    metaOutboundConnectors.add(metaOutboundConnector);
+                }
+            }
+            return metaOutboundConnectors;
+        } catch (IdentityProviderManagementException e) {
+            throw handleIdPException(e, Constants.ErrorMessage.ERROR_CODE_ERROR_RETRIEVING_META_CONNECTORS, null);
+        }
+    }
+
+    /**
+     * Get meta information about a specific outbound provisioning connector supported by the IDPs.
+     *
+     * @param id Outbound Provisioning Connector ID.
+     * @return MetaOutboundConnector.
+     */
+    public MetaOutboundConnector getMetaOutboundConnector(String id) {
+
+        String connectorName = base64URLDecode(id);
+        MetaOutboundConnector connector = null;
+        try {
+            ProvisioningConnectorConfig[] connectorConfigs = IdentityProviderServiceHolder.getIdentityProviderManager()
+                    .getAllProvisioningConnectors();
+            if (connectorConfigs != null) {
```

Comment:
The above method has used `ArrayUtils.isNotEmpty` to check if the array is empty. Why is this different?